File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []
        openb = {'[':']', '{':'}', '(':')'}
        for char in s:
            if char in openb:
                st.append(char)
            else:
                if st == []:
                    return False
                elif openb[st[-1]]!=char:
                    return False
                else:
                    st.pop()
        if st==[]:
            return True
        return False

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
#include<stack>
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        st.push('#');
        for(char c:s){
            if(c=='(' || c=='[' || c=='{'){
                st.push(c);
            }
            else if(c==')'){
                if(st.top()!='('){
                    return false;
                }
                else{
                    st.pop();
                }
            }
            else if(c==']'){
                if(st.top()!='['){
                    return false;
                }
                else{
                    st.pop();
                }
            }
            else if(c=='}'){
                if(st.top()!='{'){
                    return false;
                }
                else{
                    st.pop();
                }
            }
        }
        if(st.top()=='#'){
            return true;
        }
        return false;
    }
};